src/Symfony/Component/Console/Command/Command.php

```diff
@@ -628,4 +633,46 @@ private function validateName($name)
             throw new InvalidArgumentException(sprintf('Command name "%s" is invalid.', $name));
         }
     }
+
+    /**
+     * Locks a command.
+     *
+     * @param bool $silent
+     *
+     * @return bool
+     *
+     * @throws RuntimeException
+     */
+    protected function lock($silent = false)
```

Comment:
What about CRONs ? If a command runs as a CRON very frequently and it takes a lot of time to run then it would generate lots of errors in the logs, no ?